// Copyright 2019 Tulip Solutions B.V.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// [#protodoc-title: Common types]

syntax = "proto3";
package tulipsolutions.api.common;

option java_multiple_files = true;
option java_package = "nl.tulipsolutions.api.common";
option go_package = "github.com/tulipsolutions/tecl/tulipsolutions/api/common/orders";

enum Market {
    INVALID_MARKET = 0;
    // [#example-value-ok:]
    BTC_EUR = 1;
    // [#example-value-ok:]
    BTC_USD = 2;
}

enum Currency {
    INVALID_CURRENCY = 0;
    // [#example-value-ok:]
    BTC = 1;
    // [#example-value-ok:]
    USD = 2;
    // [#example-value-ok:]
    EUR = 3;
}

enum Side {
    INVALID_SIDE = 0;
    // [#example-value-ok:]
    BUY = 1;
    // [#example-value-ok:]
    SELL = 2;
}
